src/main/java/StreamAPI1/ListOperations.java

```
@@ -0,0 +1,70 @@
+package StreamAPI1;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class ListOperations {
+    public static int sumOfEvenNumbers(List<Integer> numbers) {
+        return numbers.stream()
+                .filter(n -> n % 2 == 0)
+                .reduce(0, Integer::sum);
+    }
+
+    public static int findMax(List<Integer> numbers) {
+        Optional<Integer> maxNumber = numbers.stream()
+                .max(Integer::compareTo);
+        if (maxNumber.isPresent()) {
+            return maxNumber.get();
+        } else {
+            return 0;
```

Мы не можем возвращать 0 отсюда, т.к. выглядит, будто это максимум, который мы нашли в списке. А список может быть пустым. Поэтому либо Optional возвращаем, либо кидаем исключение

И еще используй методы Optional, а не if-else. Например, orElse, orElseThrow

---

agreed) done

---

Yeah, returning 0 from the collection was a bad idea. If it is empty, then it is better to throw an exception implying that the list is empty and nothing to return